src/Cone3d.elm

```diff
@@ -0,0 +1,391 @@
+cone's base point towards its tip.
+-}
+axis : Cone3d units coordinates -> Axis3d units coordinates
+axis (Types.Cone3d cone) =
+    cone.axis
+
+
+{-| Get the axial direction of a cone.
+-}
+axialDirection : Cone3d units coordinates -> Direction3d coordinates
+axialDirection cone =
+    Axis3d.direction (axis cone)
+
+
+{-| Get the base radius of a cone.
+-}
+radius : Cone3d units coordinates -> Quantity Float units
+radius (Types.Cone3d cone) =
+    cone.radius
+
+
+{-| Get the base diameter of a cone (twice its radius).
+-}
+diameter : Cone3d units coordinates -> Quantity Float units
+diameter cone =
+    Quantity.twice (radius cone)
+
+
+{-| Get the overall length of a cone.
+-}
+length : Cone3d units coordinates -> Quantity Float units
+length (Types.Cone3d cone) =
+    cone.length
+
+
+{-| Get the base point of a cone. This is the center point of the circle
+that forms the base of the cone.
+-}
+basePoint : Cone3d units coordinates -> Point3d units coordinates
+basePoint cone =
+    Axis3d.originPoint (axis cone)
+
+
+{-| Get the tip point of a cone.
+-}
+tipPoint : Cone3d units coordinates -> Point3d units coordinates
+tipPoint cone =
+    Point3d.along (axis cone) (length cone)
+
+
+{-| Get the circle at the base of a cone. The axial direction of this
+circle will be the _reverse_ of the axial direction of the cone itself (the
+circle axial direction will point backwards/outwards).
+-}
+base : Cone3d units coordinates -> Circle3d units coordinates
+base cone =
+    Circle3d.withRadius (radius cone)
+        (Direction3d.reverse (axialDirection cone))
+        (basePoint cone)
+
+
+{-| Get the plane at the base of a cone. The normal direction of this
+plane will be the same as the axial direction of the cone itself.
+-}
+basePlane : Cone3d units coordinates -> Plane3d units coordinates
+basePlane cone =
+    Plane3d.through (basePoint cone) (axialDirection cone)
+
+
+{-| Get the volume of a cone.
+-}
+volume : Cone3d units coordinates -> Quantity Float (Cubed units)
+volume cone =
+    Quantity.multiplyBy pi (Quantity.squared (radius cone))
+        |> Quantity.times (length cone)
+        |> Quantity.divideBy 3
+
+
+{-| Get the minimal bounding box containing a given cone.
+-}
+boundingBox : Cone3d units coordinates -> BoundingBox3d units coordinates
+boundingBox cone =
+    BoundingBox3d.union
+        (Circle3d.boundingBox (base cone))
+        (BoundingBox3d.singleton (tipPoint cone))
+
+
+{-| Check if a cone contains a given point.
+-}
+contains : Point3d units coordinates -> Cone3d units coordinates -> Bool
+contains point cone =
+    let
+        coneAxis =
+            axis cone
+
+        axialDistance =
+            Point3d.signedDistanceAlong coneAxis point
+
+        radialDistance =
+            Point3d.distanceFromAxis coneAxis point
+
+        radiusAtPoint =
+            Quantity.multiplyBy (Quantity.ratio axialDistance (length cone)) (radius cone)
```

Comment:
Wait, isn't this logic reversed? Since axial distance is zero at the base of the cone, I think this should probably be

```elm
radiusAtPoint =
    Quantity.interpolateFrom (radius cone) Quantity.zero <|
        (Quantity.ratio axialDistance (length cone))
```

Reply:
Oh, good catch! Indeed when `axialDistance` is `0`, this would produce `0` radius.

Reply:
Fixed!